discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -751,17 +752,14 @@ def _ha_not_none(self):
             and self._config.platform != consts.Platforms.NONE
         )
 
-    def download_image(self, iso_download_path=None):
-        if self._config.is_static_ip:
-            static_network_config = static_network.generate_static_network_data_from_tf(self.nodes.controller.tf_folder)
-        else:
-            static_network_config = None
-
-        self.generate_and_download_infra_env(
-            iso_download_path=iso_download_path or self._config.iso_download_path,
-            iso_image_type=self._config.iso_image_type,
-            static_network_config=static_network_config,
-        )
+    def download_image(self, iso_download_path: str = None) -> Path:
+        if self._infra_env is None:
+            log.warning("No infra_env found. Generating infra_env and downloading ISO")
```

Comment:
I don't think finding or not finding the infraEnv should change this behavior.
It's still valid to call generate for v1 and update the infraEnv in v2.

Reply:
Not sure I fully understand, but I don't think it can hurt :man_shrugging:

Reply:
It does generate the image. As far as I know, creating infra_env generates also an ISO